pkg/util/steps/condition.go

```diff
@@ -66,9 +71,48 @@ func (c conditionStep) run(ctx context.Context, log *logrus.Entry) error {
 		log.Warnf("step %s failed but has configured 'fail=%t'. Continuing. Error: %s", c, c.fail, err.Error())
 		return nil
 	}
+	if err == wait.ErrWaitTimeout {
+		return c.enrichConditionTimeoutError()
+	}
 	return err
 }
 
+func (c conditionStep) enrichConditionTimeoutError() error {
+	funcNameParts := strings.Split(FriendlyName(c.f), ".")
+	funcName := strings.Replace(funcNameParts[len(funcNameParts)-1], "-fm", "", 1)
+
+	var message string
+	switch funcName {
+	case "apiServersReady":
+		message = "Kube API has not initialised successfully and is unavailable"
+	case "minimumWorkerNodesReady":
+		message = "minimum number of worker nodes have not been successfully created"
+	case "operatorConsoleExists":
+		message = "Console Cluster Operator has failed to initialize successfully"
+	case "operatorConsoleReady":
+		message = "Console Cluster Operator has not started successfully"
+	case "clusterVersionReady":
+		message = "Cluster Verion is not reporting status as ready"
+	case "ingressControllerReady":
+		message = "Ingress Cluster Operator has not started successfully"
+	case "aroDeploymentReady":
+		message = "ARO Cluster Operator has failed to initialize successfully"
+	case "ensureAROOperatorRunningDesiredVersion":
+		message = "ARO Cluster Operator is not running desired version"
+	case "hiveClusterDeploymentReady":
+		message = "Cluster Hive deployment is unsuccessful and is unavailable"
+	case "hiveClusterInstallationComplete":
+		message = "Cluster Hive Installation is unsuccessful"
```

Comment:
I'd like to see this as a map rather than a switch. e.g.

```
const TimeoutConditionErrors = map[string]string{
   "apiServersReady": "Kube API has not initialised successfully and is unavailable",
  "minimumWorkerNodesReady": ""minimum number of worker nodes have not been successfully created"
  // ...
}

message, exists := TimeoutConditionErrors[funcName]
if exists == false {
  return errors.New("timed out waiting for the condition")
}
```